src/gen_mod.erl

```diff
@@ -197,7 +205,7 @@ stop_module_for_host_type(HostType, Module) ->
         {wait, Process} ->
             wait_for_process(Process);
         _ ->
-            ok
+            run_for_instrumentation(HostType, fun mongoose_instrument:tear_down/1, Module)
```

Comment:
If I understand correctly `run_for_instrumentation/3` would be called only if we don't wait for any process. Is it the desired behaviour?

Reply:
No it's not intentional, I just put it here temporarily and never fixed it.